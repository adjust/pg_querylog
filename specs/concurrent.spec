setup
{
	create schema querylog;
	create extension pg_querylog schema querylog;
	create table one(a int);
}

teardown
{
	drop schema querylog cascade;
}

session "s1"
step "s1init" {
	begin work;
	lock table one in access exclusive mode;
}
step "getr" { select query, params from querylog.running_queries; }
step "getq" { select query, params from querylog.running_queries; }
step "s1end" { commit work; }

session "s2"
step "q1" {
	prepare p1 as select $1, * from one;
	execute p1(1);
}

session "s3"
step "q2" {
	select 1;
}

permutation "s1init" "q1" "q2" "getr" "getq" "s1end"
