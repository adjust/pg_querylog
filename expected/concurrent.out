Parsed test spec with 3 sessions

starting permutation: s1init q1 q2 getr getq s1end
step s1init: 
	begin work;
	lock table one in access exclusive mode;

step q1: 
	prepare p1 as select $1, * from one;
	execute p1(1);
 <waiting ...>
step q2: 
	select 1;

?column?       

1              
step getr: select query, params from querylog.running_queries;
query          params         

select query, params from querylog.running_queries;               
step getq: select query, params from querylog.running_queries;
query          params         

select query, params from querylog.running_queries;               
step s1end: commit work;
step q1: <... completed>
?column?       a              

